File: 백준/Silver/3273. 두 수의 합/두 수의 합.js
let fs = require('fs');
let [read1, read2, read3] = fs.readFileSync('/dev/stdin').toString().trim().split('\n');
const n = Number(read1);
const x = Number(read3);
const arr = read2.split(' ').map((value) => Number(value)).sort((a, b) => a- b);

let start = 0;
let end = n - 1;
let answer = 0;

while(start < end){
    const temp = arr[start] + arr[end];
    if( temp === x){
        start++;
        answer++;
    } else if(temp < x){
        start++;
    } else if(temp > x){
        end--;
    }
}
console.log(answer);

File: 백준/Silver/3273. 두 수의 합/두 수의 합.py
n = int(input())

arr = sorted(list(map(int, input().split())))

x = int(input())
i = 0
j = n - 1

cnt = 0

while i < j:
    if arr[i] + arr[j] == x:
        cnt += 1
        i += 1
        j -= 1
    elif arr[i] + arr[j] < x:
        i += 1
    elif arr[i] + arr[j] > x:
        j -= 1
print(cnt)
    